### PR TITLE
[C-2842] Improve playlist image generation

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Improves playlist image generation for all playlist track actions: `add` `remove` and `reorder`
- Also fixes playlist_library issue where local user data wouldn't update with latest value.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2842-improve-image-generation

File: packages/web/src/components/create-playlist/PlaylistForm.tsx [hunk 1]
```diff
@@ -141,11 +141,13 @@ const PlaylistForm = ({
   )
 
   const handleRemoveArtwork = useCallback(async () => {
-    const { file, url } = await onGeneratePlaylistArtwork()
+    const artwork = await onGeneratePlaylistArtwork()
+    console.log('the artwork?', artwork)
```
Comment (Kyle-Shanks) [inline packages/web/src/components/create-playlist/PlaylistForm.tsx:145]: log

File: packages/common/src/utils/updatePlaylistArtwork.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    } else if (tracks.length === 3) {
+      tracksForImage = [...tracks, actions.added]
+    }
+  } else if (actions.removed) {
+    const { removed } = actions
+    const removedIndex = tracks.findIndex(
+      (track) => track.track_id === removed.track_id
+    )
+    if (removedIndex === -1) {
+      // continue
```
Comment (Kyle-Shanks) [inline packages/common/src/utils/updatePlaylistArtwork.ts:47]: what case is this?

Comment (Kyle-Shanks) [reply packages/common/src/utils/updatePlaylistArtwork.ts:47]: oh is this just the default value passed when nothing is removed?

**Review by Kyle-Shanks — APPROVED** [review]

File: packages/common/src/utils/updatePlaylistArtwork.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/common/src/utils/updatePlaylistArtwork.ts:47]: yeah in case we dont get a match, just gonna carry on :)

**Pull request merged**